evm/test/RateLimit.t.sol

```diff
@@ -698,4 +701,321 @@ contract TestRateLimit is Test, IRateLimiterEvents {
             assertEq(inboundLimitParams.lastTxTimestamp, sendAgainTime);
         }
     }
+
+    // helper functions
+    function setupToken() public returns (address, address, DummyToken, uint8) {
+        address user_A = address(0x123);
+        address user_B = address(0x456);
+
+        DummyToken token = DummyToken(nttManager.token());
+
+        uint8 decimals = token.decimals();
+        assertEq(decimals, 18);
+
+        return (user_A, user_B, token, decimals);
+    }
+
+    function initializeTransceivers() public returns (ITransceiverReceiver[] memory) {
+        (DummyTransceiver e1, DummyTransceiver e2) =
+            TransceiverHelpersLib.setup_transceivers(nttManager);
+
+        ITransceiverReceiver[] memory transceivers = new ITransceiverReceiver[](2);
+        transceivers[0] = e1;
+        transceivers[1] = e2;
+
+        return transceivers;
+    }
+
+    // transfer tokens from user_A to user_B
+    // this consumes capacity on the outbound side
+    // send tokens from user_B to user_A
+    // this consumes capacity on the inbound side
+    // send tokens from user_A to user_B
+    // this should consume capacity on the outbound side
+    // and backfill the inbound side
+    function testFuzz_CircularFlowBackFilling(uint64 mintAmt, uint64 transferAmt) public {
+        vm.assume(transferAmt > 0 && transferAmt < mintAmt);
+
+        (address user_A, address user_B, DummyToken token, uint8 decimals) = setupToken();
+
+        TrimmedAmount memory mintAmount = TrimmedAmount(mintAmt, 8);
+        token.mintDummy(address(user_A), mintAmount.untrim(decimals));
+        nttManager.setOutboundLimit(mintAmount.untrim(decimals));
+
+        // transfer 10 tokens
+        vm.startPrank(user_A);
+
+        TrimmedAmount memory transferAmount = TrimmedAmount(transferAmt, 8);
+        token.approve(address(nttManager), type(uint256).max);
+        // transfer tokens from user_A -> user_B via the nttManager
+        nttManager.transfer(
+            transferAmount.untrim(decimals), chainId, toWormholeFormat(user_B), false, new bytes(1)
```

Comment:
Why did you choose to use the `TrimmedAmount` struct to hold these amounts? Doesn't that mean we'll never have an amount > `type(uint64).max` as a `transfer` input?

Reply:
i think this is fine for this test (its not really meant to test the queueing logic). can test that more explicitly in a separates test. the amount transferred should be less than the outbound limit. `setOutboundLimit` trims the `uint256` value, which if greater than u64MAX, reverts on `AmountTooLarge.` using trimmed amounts makes the inputs more explicit. to improve this slightly, we could uint256 values as inputs and then trim (and check the revert case).

Reply:
Yeah so my thinking around this wasn't for testing the queuing logic (I agree we don't need to test that here), but more that the trimming works properly and reverts when we expect.

Rather than branching in this fuzz test we could have another fuzz test that always reverts for an expected bounded set of input values?

Reply:
yea makes sense to include some more negative tests - will add that in